### PR TITLE
PTEMSVT-241 V2 Benefits Logic

### PR DESCRIPTION
**Pull request by ghost**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This PR adds more benefits logic for V2 Benefits eligibility questionnaire.

## Testing done

Manual and unit test

## Screenshots
Screen Shot 2024-10-28 at 8 31 43 AM

Screen Shot 2024-10-28 at 8 31 09 AM

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/benefit-eligibility-questionnaire/constants/benefits.js [hunk 1]
```diff
@@ -435,6 +440,203 @@ export const BENEFITS_LIST = [
     learnMoreURL: 'https://www.va.gov/careers-employment/vetsuccess-on-campus/',
     applyNowURL: '',
   },
+  {
+    name: 'Disability housing',
```
Comment (vetskrieg) [inline src/applications/benefit-eligibility-questionnaire/constants/benefits.js:444]: Should this be "Disability housing grant"?

**Review by vetskrieg — DISMISSED** [review]

File: src/applications/benefit-eligibility-questionnaire/constants/benefits.js [hunk 2]
```diff
@@ -435,6 +440,203 @@ export const BENEFITS_LIST = [
...
+    category: categories.HOUSING,
+    id: 'DHS',
+    description:
+      "We offer housing grants for for Veterans and service-connected disabilities so they can buy or change a home to meet their needs and live more independentaly. Find out if you're eligible for a disability housing grant and how to apply.",
+    isTimeSensitive: false,
+    mappings: {
+      [mappingTypes.GOALS]: [
+        goalTypes.FINANCIAL,
+        goalTypes.PLAN,
+        goalTypes.UNDERSTAND,
```
Comment (vetskrieg) [inline src/applications/benefit-eligibility-questionnaire/constants/benefits.js:454]: goalTypes.UNDERSTAND is duplicated here.

**Review by vetskrieg — COMMENTED** [review]
Can you re-order the goalTypes to match the Figma? DEGREE comes after COMMEMORATE.

**Review by vetskrieg — COMMENTED** [review]
```
export const goalTypeLabels = Object.freeze({
   LEGACY: 'Carry out legacy planning',
   COMMEMORATE: 'Commemorate service',
-  FINANCIAL: 'Get financial support ',
+  DEGREE: 'Earn a degree or certificate',^M
+  JOBS: 'Find a civilian job',^M
   HOUSING: 'Find a place to live ',
+  FINANCIAL: 'Get financial support ',^M
   FAMILY: 'Grow my family ',
   MENTAL: 'Improve my mental well-being',
   PHYSICAL: 'Improve my physical well-being',
-  DEGREE: 'Earn a degree or certificate',
-  JOBS: 'Find a civilian job',
   PLAN: 'Plan for my transition',
   CAREER_PATH: 'Set a career path',
   BUSINESS: 'Start a business',
```

**Review by ataker — DISMISSED** [review]

**Review by ataker — APPROVED** [review]

**Pull request merged**